### DISCUSSION TITLE
What is the relationship between Autorest and Kiota

### DISCUSSION THREAD

**Discussion by wtrocki** [Azure/autorest#4800] [Q&A] [answered]
I have been autorest user for quite a long time. 
Is there any relation between autorest and https://github.com/microsoft/kiota project? 
Are those projects competing in any way?

**Comment by timotheeguerin** [comment, marked as answer]
There is this issue on kiota that discusss the differences both project fits different needs https://github.com/microsoft/kiota/issues/1709.

Autorest was designed to generate SDK for Azure services and generators are really targeted for the constraints that azure SDK require but where each service is its own independent entity. Kiota on the other hand was design to work with the scale of Microsoft graph where every api are interconnected.

Note that we are not investing much more in autorest as we are migrating Azure Apis to use our new API definition language https://microsoft.github.io/typespec. Generators there also worked on to work better with some non Azure-apis.

**Reply by wtrocki** [reply]
Thank you so much
